### ISSUE TITLE
implement generate citation button

### ISSUE THREAD

**Issue by foxm** [blakearchive/archive#171]

**Comment by queryluke** [comment]
@foxm where is the citation found? Is it in the xml somewhere?

**Comment by foxm** [comment]
@queryluke no, it has to be generated from various data in a couple of formats, apa and chicago. i'll come up with some examples

**Comment by queryluke** [comment]
@nathan-rice since this is string manipulation, do you want to implement this in the Python import scripts? I.e., adding citation to the object model?

Or generate via JavaScript over the json?

**Comment by nathan-rice** [comment]
I would have to see exactly what data from the model is required to say for certain.  That being said, I expect it will probably be a better choice to pull the requisite fields from the model's JSON data, and render the citation with Javascript; this lets you update the citation generation code without re-populating the database.

**Comment by queryluke** [comment]
@foxm going to work on this soon, if you can provide those examples
